lib/Server.js

```diff
@@ -146,7 +151,7 @@ function Server(compiler, options) {
 							}
 							proxy.web(req, res, proxyOptions, function(err){
 								var msg = "cannot proxy to " + proxyOptions.target + " (" + err.message + ")";
-								this.io.sockets.emit("proxy-error", [msg]);
+								this.sockWrite("proxy-error", [msg]);
```

Comment:
Should this also be guarded with a verification that `this.sock` exists? I get a crash when this runs because `this.sock` is undefined in the `this.sockWrite` call. Perhaps that guard should simply be put inside the `sockWrite` call?